### PR TITLE
Update to I2C docs: Clarify address expectation and return values

### PR DESCRIPTION
**Pull request by brickbots**
I got caught out a bit trying to implement this driver for some I2C sensors and thoughts I'd update the docs to help others who might run into the same pitfalls

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/i2c_driver.md [hunk 1]
```diff
@@ -2,17 +2,35 @@
 
 The I2C Master drivers used in QMK have a set of common functions to allow portability between MCUs.
 
+## An important note on I2C Addresses
+
+All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
+the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
+on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
+shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`
+
```
Comment (noroadsleft) [inline docs/i2c_driver.md]:
```suggestion
All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`.
```

File: docs/i2c_driver.md [hunk 2]
```diff
@@ -2,17 +2,35 @@
...
+You can either do this on each call to the functions below, or once in your definition of the address.  For example if your device has an address of `0x18`:
+
+`#define MY_I2C_ADDRESS 0x18 << 1`
+
+See: 
+
+https://www.robot-electronics.co.uk/i2c-tutorial 
+
+For more information about I2C addressing and other technial details.
```
Comment (noroadsleft) [inline docs/i2c_driver.md]:
```suggestion
See https://www.robot-electronics.co.uk/i2c-tutorial for more information about I2C addressing and other technical details.
```

File: docs/i2c_driver.md [hunk 3]
```diff
@@ -2,17 +2,34 @@
...
+## An important note on I2C Addresses
+
+All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
+the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
+on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
+shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`.
+
+You can either do this on each call to the functions below, or once in your definition of the address.  For example if your device has an address of `0x18`:
+
+`#define MY_I2C_ADDRESS 0x18 << 1`
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
`#define MY_I2C_ADDRESS (0x18 << 1)`
```

File: docs/i2c_driver.md [hunk 4]
```diff
@@ -24,6 +41,15 @@ All the above functions, except `void i2c_init(void);` return the following trut
 |-1             |Operation failed.                                  |
 |-2             |Operation timed out.                               |
 
+These are defined in `i2c_master.h` and it is suggested you use this definitions for future compatibility if
+they change.
+
+```
+#define I2C_STATUS_SUCCESS (0)
+#define I2C_STATUS_ERROR (-1)
+#define I2C_STATUS_TIMEOUT (-2)
```
Comment (fauxpark) [inline docs/i2c_driver.md]: IMO these names should be part of the above table.

**Comment by brickbots** [comment]
Thanks @fauxpark and @noroadsleft for your great suggestions!  I've added a commit to implement them :-)

File: docs/i2c_driver.md [hunk 5]
```diff
@@ -2,27 +2,44 @@
...
-|Return Value   |Description                                        |
-|---------------|---------------------------------------------------|
-|0              |Operation executed successfully.                   |
-|-1             |Operation failed.                                  |
-|-2             |Operation timed out.                               |
+|Return Constant     |Value |Description                                 |
+|--------------------|---------------------------------------------------|
+|I2C_STATUS_SUCCESS  |  0   |Operation executed successfully.            |
+|I2C_STATUS_ERROR    | -1   |Operation failed.                           |
+|I2C_STATUS_TIMEOUT  | -2   |Operation timed out.                        |
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
|Return Constant     |Value|Description                     |
|--------------------|-----|--------------------------------|
|`I2C_STATUS_SUCCESS`|0    |Operation executed successfully.|
|`I2C_STATUS_ERROR`  |-1   |Operation failed.               |
|`I2C_STATUS_TIMEOUT`|-2   |Operation timed out.            |
```

File: docs/i2c_driver.md [hunk 6]
```diff
@@ -2,27 +2,44 @@
...
+
+These are defined in `i2c_master.h` and it is suggestd you use the definitions for future compatibility if
+they change.
+
```
Comment (fauxpark) [inline docs/i2c_driver.md]:
```suggestion
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**